adagrams/game.py

```diff
@@ -1,11 +1,116 @@
+import random
+
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+LETTER_SCORE = {
+    'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10
+}
+
 def draw_letters():
-    pass
+
+    user_letters = []
+    copy_letter_pool = dict(LETTER_POOL)
+    print(copy_letter_pool)
+    while len(user_letters) < 10:
+        random_letter = random.choice(list(copy_letter_pool.keys()))
+        user_letters.append(random_letter)
+        copy_letter_pool.update({random_letter:copy_letter_pool[random_letter] - 1}) # Reduce value of letter by 1
+        if copy_letter_pool[random_letter] == 0: # Delete any letter from dict with value of 0
+            del copy_letter_pool[random_letter]
+
+    return user_letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    aCopy = list(letter_bank)
+    for letter in word:
+        if letter.upper() in aCopy:
```

Comment:
The `in` operator on lists has linear time complexity on top of the first loop that's running on loop 79. You can optimize this here by using letter_bank to create a frequency dictionary where the keys are the letters and the values are the frequency of the letter. For example: letter_frequency = {"a": 1, "e": 2}

Then when you check if a letter is a key of letter_frequency, you reduce your time complexity to O(1).
```python
for letter in word:
     if letter in letter_frequency:
```

Take care to use `key in dict` syntax. If you do `key in dict.keys()` you'd reproduce O(n) time complexity since under the hood Python creates a list of keys when you call .keys() and the in operator for lists is linear.